### PR TITLE
Update accounts model to represent account profile data

### PR DESCRIPTION
**Pull request by tibetsprague**
Don’t allow anyone to update the account without passing along a signed message proving they own their ethereal account address.

### CODE REVIEW

File: common/models/account.js [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
 module.exports = function(Account) {
-  Account.greet = function(msg, cb) {
-    process.nextTick(function() {
-      msg = msg || 'hello';
-      cb(null, 'Sender says ' + msg + ' to receiver');
-    });
-  };
+
+  Account.observe('before save', function(ctx, next) {
+    const text = "Please sign in to Alchemy";
```
Comment (tibetsprague) [inline common/models/account.js]: I don't think its very secure to just use this same message for everyone and have it in the codebase like this, but not sure what another easy solution would be for now

Comment (tsuberim) [reply common/models/account.js]: Why isn't it secure to use the same message for all accounts?

File: common/models/account.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
+    const msg = ethUtil.bufferToHex(Buffer.from(text, 'utf8'));
+    const recoveredAddress = sigUtil.recoverPersonalSignature({ data: msg, sig: ctx.instance.signature });
```
Comment (tsuberim) [inline common/models/account.js]: is `ctx.instance.signature` saved in the db?

Comment (tibetsprague) [reply common/models/account.js]: It's not, but it is saved in local storage on the device so you dont need to sign on that device again

File: common/models/account.js (hunk 1 above)
Comment (tibetsprague) [reply common/models/account.js]: hmm, yeah nevermind

**Pull request merged**